File: 1769-minimum-number-of-operations-to-move-all-balls-to-each-box/1769-minimum-number-of-operations-to-move-all-balls-to-each-box.cpp
class Solution {
public:
    vector<int> minOperations(string boxes) {
        int n = boxes.size(), j = 0;
        vector<int> res(n, 0);

        for(int i = 0; i < n; i++) {
            if(boxes[i] == '1') {
                j = 0;
                while(j < n) {
                    res[j] += abs(j-i);
                    j++;
                }
            }
        }

        return res;
    }
};

File: 1769-minimum-number-of-operations-to-move-all-balls-to-each-box/1769-minimum-number-of-operations-to-move-all-balls-to-each-box.js
/**
 * @param {string} boxes
 * @return {number[]}
 */
var minOperations = function(boxes) {
    let n = boxes.length, arr = Array(n).fill(0), cSum = 0, cVal = 0;

    for(let i = 0; i < n; i++) {
        arr[i] += cSum;
        cVal += +boxes[i];
        cSum = cSum + cVal;
    }

    cSum = 0, cVal = 0;
    for(let i = n - 1; i >= 0; i--) {
        arr[i] += cSum;
        cVal += +boxes[i];
        cSum = cSum + cVal;
    }

    return arr;
};